lib/pay/stripe/subscription.rb

```diff
@@ -36,12 +36,15 @@ def self.sync(subscription_id, object: nil, name: Pay.default_product_name)
           stripe_account: owner.stripe_account,
           trial_ends_at: (object.trial_end ? Time.at(object.trial_end) : nil)
         }
-
-        # Subscriptions cancelling in the future
-        attributes[:ends_at] = Time.at(object.current_period_end) if object.cancel_at_period_end
-
-        # Fully cancelled subscription
-        attributes[:ends_at] = Time.at(object.ended_at) if object.ended_at
+        if object.ended_at
+          # Fully cancelled subscription
+          attributes[:ends_at] = Time.at(object.ended_at)
+        elsif object.cancel_at_period_end
+          # Subscriptions cancelling in the future
+          attributes[:ends_at] = Time.at(object.current_period_end)
+        elsif object.cancel_at.nil?
+          attributes[:ends_at] = nil
+        end
```

Comment:
I'm thinking this might need to be changed so we handle the `cancel_at` column properly.

```ruby
elsif object.cancel_at
  attributes[:ends_at] = Time.at(object.cancel_at)
else
  attributes[:ends_at] = nil
end
```

This would then mark the subscription `ends_at` anytime there's an ended_at, cancel_at, or cancel_at_period_end. Resetting to nil if not.

Reply:
that was what I was thinking too, but didn't know if there were special meanings here or there.